spec/requests/api/v2/app_groups_spec.rb

```diff
@@ -1,5 +1,79 @@
 require 'rails_helper'
 
+RSpec.describe 'Deactivated App Group by Cluster Name', type: :request do
+  let(:headers) do
+    { 'ACCEPT' => 'application/json', 'HTTP_ACCEPT' => 'application/json' }
+  end
+
+  before(:all) do
+    @access_token = 'E8cjTcUKucBxRs-XV_EvosUgMsytLGVBhmD_b6KpOVbYnhX3THgVtLMXOHS606Us'
+    @ext_app = create(:ext_app, access_token: @access_token)
+  end
+
+  describe 'Deactivated App Group by Cluster Name' do
+    it 'should deactivated the app group' do
+      # Create a App Group
+      app_group = create(:app_group)
+      helm_infrastructure = create(
+        :helm_infrastructure,
+        app_group: app_group,
+        cluster_name: "test-delete",
+        status: HelmInfrastructure.statuses[:active],
+        provisioning_status: HelmInfrastructure.provisioning_statuses[:deployment_finished]
+      )
+
+      # Send a DELETE request to the delete action
+      post api_v2_deactivated_by_cluster_name_path(cluster_name: helm_infrastructure.cluster_name),params:{access_token: @access_token}, headers: headers
+
+      # Expect a successful response
+      expect(response.status).to eq(200)
+
+      # Parse the JSON response
+      json_response = JSON.parse(response.body)
+
+      # Expect the response to indicate success
+      expect(json_response['success']).to eq(true)
+      expect(json_response['message']).to eq('App Group deactivated successfully')
+
+      # Verify that the Helm Infrastructure has been deleted
+      expect(HelmInfrastructure.find_by(cluster_name: helm_infrastructure.cluster_name).provisioning_status).to include('DELETE')
```

Comment:
need to use a match, at the start it will `DELETE_STARTED` after a while, it will update to `DELETED`. we can use the sleep or wait function. first one check to match("DELETE_STARTED") and then put sleep function like 5s and add to match("DELETED")